src/verify.c

```diff
@@ -474,6 +478,29 @@ static void add_missing_files(struct manifest *official_manifest)
 		if ((ret != 0) || hash_needs_work(file, local.hash)) {
 			file_not_replaced_count++;
 			printf("\n\tnot fixed\n");
+
+			if (cmdline_option_install && no_freespace_flag == false) {
+				fs_free = get_available_space(path_prefix);
+				string_or_die(&original, "%s/staged/%s", state_dir, file->hash);
+				if (stat(original, &st) == 0) {
+					if ((st.st_size * 1.1) > fs_free) {
+						if (fs_free >= 0) {
+							fprintf(stderr, "\tWarning: File to install (%s) too large by %ldK.\n", file->filename, (st.st_size - fs_free) / 1000);
+							/* set flag to skip checking space on the second failure, assume we're still out of space */
+							no_freespace_flag = true;
+						} else if (fs_free < 0) {
+							fprintf(stderr, "\tWarning: Unable to determine free space on filesystem.\n");
+						}
+						fprintf(stderr, "\tContinuing with install...\n");
```

Comment:
Move this after the below `else` so it is consistently printed after each `Unable to determine free space...` warning.